site/en/docs/extensions/mv3/content_scripts/index.md

```diff
@@ -171,42 +178,53 @@ They can include JavaScript files, CSS files, or both. All auto-run content scri
   </tbody>
 </table>
 
-{% if false %}
-
 ### Inject with dynamic declarations {: #dynamic-declarative }
 
-{% Aside 'caution' %}
-
-This feature is not yet fully supported. It is currently in dev and is also available in Chrome
-Canary.
-
-{% endAside %}
-
-You should use dynamic declarations in the following cases:
+Dynamic content scripts are useful for situations when the match patterns for a content scripts are
+not well known or when content scripts should not always be injected on known hosts.
 
-- When the host is not well known
-- The script may need to be added/removed from a known host
+Introduced in Chrome 96, dynamic declarations are similar to [static
+declarations][header-cs-static], but the content script object is registered with Chrome using
+methods on the Scripting API rather than in [manifest.json][doc-manifest]. In addition to
+[registering][api-register-cs] content scripts, the Scripting API also allows extension developers
+to [get a list of][api-get-registered-cs], [update][api-update-cs], and [remove][api-remove-cs]
+content script registrations.
```

Comment:
Can you be more specific? Without more context I'm not sure how to resolve the confusion.

The purpose of this sentence is to outline the set of available operations a developer can perform when when working with dynamic content scripts.

Reply:
"To get a list of update and remove content script registrations.'

Grammatically, I would expect 'updated and removed'. I tried following the links to sort this. It didn't clarify anything for me. 

By the way, there's no URL for `api-remove-cs`. This is why I favor keeping URLs in line.

Reply:
For the sake of discussion I'll be more explicit here than I was in the current draft.

Content scripts can be declared either statically or dynamically. Static registration is handled in `manifest.json` and dynamic registration is handled at runtime via the Scripting API. The Scripting API provides several methods to interact with dynamic content script registrations:

| method | description |
| --- | --- |
| `registerContentScripts()` | Create a new dynamic content script registration |
| `getRegisteredContentScripts()` | Retrieve a list of existing registrations. Once you have the current list of registrations, you can use other methods to edit or completely remove a registration. |
| `updateContentScripts()` | Edit one or more existing dynamic content script registrations. |
| `unregisterContentScripts()` | Delete an existing dynamic content script registration. |

This sentence attempts to list what can be done with these methods without directly listing the methods themselves. I didn't include registration in this sentence because I already mentioned in it in the previous one.

I felt like this construction was a bit weird when I first drafted it, but I was hoping that that was just in my head. After this discussion, I think it would be best to either rewrite this passage to more clearly communicate this idea or  avoid listing dynamic content script operations here entirely.

Reply:
> By the way, there's no URL for `api-remove-cs`. This is why I favor keeping URLs in line.

I didn't know you had a preference one way or the other. I'm happy to reassess this practice; it mostly came from the fact that the tool we used to migrate content from the Chromium repository used footer links rather than inline links. Personally, I prefer footer links as I find that inline links make Markdown harder to read, but that's just one of many considerations we should weigh.

Inline links might help us fix situations where a link is completely missing, but they don't necessarily help ensure links are correct or that the link points to an extant page. I'd love to have an automated tool that visits linked pages and checks for 30X redirects, 404s, 503s, etc. Better still, if this were part of our build and deploy process it could reject PRs that contain invalid links.

Reply:
I've fixed the broken link, and as discussed in our call, switched this to use a bulleted list for clarity.